contract/r/gnoswap/staker/v1/reward_calculation_incentives.gno

```diff
@@ -135,56 +141,26 @@ func (self *IncentivesResolver) calculateUnclaimableReward(incentiveId string) i
 	}
 
 	timeDiff := int64(0)
+	startTimestamp := incentive.StartTimestamp()
+	endTimestamp := incentive.EndTimestamp()
 
-	// Find unclaimable periods that end before or at incentive start
-	self.UnclaimablePeriods().ReverseIterate(0, incentive.StartTimestamp(), func(key int64, value any) bool {
-		startTimestamp := key
-		endTimestamp, ok := value.(int64)
+	self.UnclaimablePeriods().Iterate(0, endTimestamp, func(key int64, value any) bool {
```

Comment:
The `calculateUnclaimableReward` function is implemented in a way that iterates through all unclaimable intervals. As the interval grows longer, the cost could increase significantly. We need to find a way to reduce the iteration range.